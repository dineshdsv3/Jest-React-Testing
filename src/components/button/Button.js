import React from 'react';

const Button = ({ label }) => {
	return (
		<button data-testid="button" className="button-style">
			Hi {label}
		</button>
	);
};

export default Button;
