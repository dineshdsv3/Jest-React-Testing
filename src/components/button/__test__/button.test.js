import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../Button';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer';

afterEach(cleanup);
it('Renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<Button />, div);
});

it('Renders button correctly', () => {
	const { getByTestId } = render(<Button label="click me" />);
	expect(getByTestId('button')).toHaveTextContent('click me');
});

it('Renders button correctly', () => {
	const { getByTestId } = render(<Button label="me" />);
	expect(getByTestId('button')).toHaveTextContent('me');
});

it('Matches Snapshot', () => {
	const tree = renderer.create(<Button label="click me" />).toJSON();
	expect(tree).toMatchSnapshot();
});
