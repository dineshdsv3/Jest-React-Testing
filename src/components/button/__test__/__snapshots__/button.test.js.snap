// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<button
  className="button-style"
  data-testid="button"
>
  Hi 
  click me
</button>
`;
